scripts/check_fqcn.py

```diff
@@ -0,0 +1,130 @@
+import sys
+import urllib.request
+from pathlib import Path
+
+from ansible.errors import AnsibleParserError
+from ansible.inventory.manager import InventoryManager
+from ansible.parsing.dataloader import DataLoader
+from ansible.playbook.task import Task
+from ansible.vars.manager import VariableManager
+
+import ruamel.yaml
+import ruamel.yaml.util
+
+yaml = ruamel.yaml.YAML()
+
+Task._validate_attributes = lambda *args: None
+
+
+def get_tasks(obj):
+    """Get each task."""
+    if isinstance(obj, list):
+        for o in obj:
+            yield from get_tasks(o)
+    else:
+        found = False
+        for key in ('tasks',
+                    'pre_tasks',
+                    'post_tasks',
+                    'handlers',
+                    'block',
+                    'rescue',
+                    'always'):
+            if obj.get(key):
+                found = True
+                yield from get_tasks(obj.get(key))
+        if not found:
+            yield obj
+
+
+def check_tasks(file):
+    """Parse a tasks file."""
+    print(file)
+    errors = []
+    with open(file) as f:
+        data = yaml.load(f)
+
+    # If data is None it means is not a valid YAML file
+    if data:
+        for ds in get_tasks(data):
+            try:
+                task = Task.load(ds,
+                                 variable_manager=variable_manager,
+                                 loader=loader)
+                route_class = ['modules',
+                               'connection',
+                               'module_utils',
+                               'cliconf',
+                               'terminal',
+                               'action',
+                               'become',
+                               'cache',
+                               'callback',
+                               'doc_fragments',
+                               'filter',
+                               'httpapi',
+                               'inventory',
+                               'lookup',
+                               'netconf',
+                               'shell',
+                               'test']
+
+                for route in route_class:
+                    module = routing[route].get(task.action,
+                                                {}).get('redirect')
+                    if module:
+                        errors.append((file, task.action, module))
+                        break
+
+                if '.' not in task.action:
+                    errors.append((file, task.action, "no dot in namespace"))
+
+            except AnsibleParserError:
+                pass
+            except Exception:
+                pass
+    return errors
+
+
+loader = DataLoader()
+inventory = InventoryManager(loader=loader, sources='localhost,')
+variable_manager = VariableManager(loader=loader, inventory=inventory)
+
+url = ("https://raw.githubusercontent.com/"
+       "ansible/ansible/devel/lib/ansible/config/"
```

Comment:
This file do not exists in 2.9 that's why I'm pulling directly from it's content in github

Reply:
Maybe we should pull the file from a tagged commit (2.10.x) rather than devel branch, In case it is moved or removed or changed unexpectedly? IIUC it's a part of an internal mechanism of Ansible, not user-facing one, so i'd assume it can change. Not a blocker though.

Reply:
Yeah, I agree - better to pin it to a released/tagged version. This file will change quite a bit over time and will likely get removed for Ansible 3.x which is on the horizon.

Reply:
done